src/js/memory/solverMemory.ts

```diff
@@ -0,0 +1,270 @@
+import { getById } from '../utilities';
+
+const position = "Position";
+const label = "Label";
+
+interface ElementData {
+	name: string;
+	stage: number;
+	value: number;
+  }
+
+export function memory_solve(e : HTMLInputElement): void {		
+	clean_form();
+
+	//only solve if checking a button
+	if (!e.checked) {
+		return;
+	}
```

Comment:
What if we are unticking a button. Like if you accidentally click a number on the next line and then turn it back off you'd want it to resolve to get back to the previous solution wouldn't you?